### PR TITLE
Add XMLString field type which validates input.

### PR DESCRIPTION
**Pull request by jcdyer**
### [TNL-5245](https://openedx.atlassian.net/browse/TNL-5245)

Add XMLString field type which validates that input is real XML.

This will be used to ensure inbound capa data is valid xml.  See reference ticket for rationale.

### Reviewers

- [x] @sanfordstudent 
- [x] @doctoryes

FYI @efischer19 @nasthagiri

### CODE REVIEW

File: xblock/__init__.py [hunk 1]
```diff
@@ -22,4 +22,4 @@ def __init__(self, *args, **kwargs):
 
 xblock.fields.XBlockMixin = XBlockMixin
 
-__version__ = "0.4.7"
+__version__ = "0.5.0"
```
Comment (doctoryes) [inline xblock/__init__.py]: Nit: It'd be nice to pull the `setup.py` version and this one from a common location, particularly considering that this version was out-of-date.

File: xblock/fields.py [hunk 2]
```diff
@@ -832,17 +834,24 @@ class String(JSONField):
...
         Leave all other characters.
         """
-        if isinstance(value, unicode):
-            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in (u'\n', u'\r', u'\t'))
-        elif isinstance(value, str):
-            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in ('\n', '\r', '\t'))
+        if isinstance(value, six.text_type):
+            new_value = u''.join(ch for ch in value if self._valid_unichar(ch))
+        elif isinstance(value, six.binary_type):
+            new_value = ''.join(ch for ch in value if self._valid_bytechar(ch))
```
Comment (doctoryes) [inline xblock/fields.py]: nice refactor!

File: xblock/test/test_fields.py [hunk 3]
```diff
@@ -244,6 +241,34 @@ def test_control_characters_filtered(self):
...
+        xml_string = self.FIELD_TO_TEST()
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, 'text')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<xml')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<xml attr=3/>')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<xml attr="3/>')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<open>')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<open>with text')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<xml/>trailing text')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<open>text</close>')
+        self.assertRaises(etree.XMLSyntaxError, xml_string.from_json, '<invalid_utf8 char="\x9e"/>')
```
Comment (doctoryes) [inline xblock/test/test_fields.py]: Both of these tests would be better off using DDT'd lists of test data.

File: xblock/__init__.py (hunk 1 above)
Comment (jcdyer) [inline xblock/__init__.py]: Done.

File: xblock/test/test_fields.py (hunk 3 above)
Comment (jcdyer) [inline xblock/test/test_fields.py]: Done.

**Comment by jcdyer** [comment]
max_pylint_violations script had non-obvious behavior: matching the max number of errors was a failure.  Should be exceeding max number is an error.

**Comment by jcdyer** [comment]
Ready for another round of review.

**Comment by jcdyer** [comment]
@cpennington Here's the PR to create the xml type.

File: xblock/VERSION.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+0.5.0
```
Comment (cpennington) [inline xblock/VERSION.txt]: This isn't backwards incompatible, right? No need to go to 0.5.0 over 0.4.13, right?

**Comment by jcdyer** [comment]
@cpennington This has been updated to validate when saving, rather than when loading, which should reduce the impact of the checks.Take a look.

**Comment by jcdyer** [comment]
@sanfordstudent @doctoryes This now does validation when serializing instead of deserializing, and only does it if enforce_type is set to True.  That way open edx installations can temporarily turn off the feature to fix bad data.

**Comment by sanfordstudent** [comment]
Still looks good to me.

**Pull request merged**